tests/neg/extension-methods.scala

```diff
@@ -15,4 +15,23 @@ object Test {
     def f2[T]: T = ???        // error: T is already defined as type T
     def f3(xs: List[T]) = ??? // error: xs is already defined as value xs
   }
-}
\ No newline at end of file
+
+  trait Chain{
+      type Tail <: Chain
+  }
+
+  // The initial clauses of right-associative extension methods are not allowed to depend on each other
+  extension [C2 <: c1.Tail](c2: C2) // Not found: c1
+    def !:[C1 <: Chain](c1: C1): c2.Tail = ???
+
+  extension [C1 <: Chain](c1: C1)
+    def ?:[C2 <: c1.Tail](c2: C2): c2.Tail = ??? // Not found: c1
```

Comment:
We didn't discuss it, but it seems to be the intended behavior (even before this feature):
https://scastie.scala-lang.org/A6pbaHSuQNucDsrOALxQjw

The reason I believe is that right associative extension methods get their parameters shuffled:
* We can probably allow the first line simply, but it looks really confusing, because things depend on parameter that come after in the definition
* The second looks normal, but is actually translated to `[C2 <: c1.Tail](c2: C2)[C1 <: Chain](c1: C1): c2.Tail`, which again has weird references

No matter what we do, the result is confusing, so not allowing either might be the best option

The thing that is missing is a friendlier error message, as it is not obvious why the second it not allowed to a user not familiar with the internals
There are also shadowing surprises like:
```Scala
val c1: Chain = "Looks shadowed, but isn't".asInstanceOf[Chain]
extension [C1 <: Chain](c1: C1)
  def ?:[C2 <: c1.Tail](c2: C2): c2.Tail = ??? // c1 is the val above, not the parameter !
```